File: 프로그래머스/lv2/12909. 올바른 괄호/올바른 괄호.py
from collections import deque
def solution(s):
    answer = True
    stack = deque()
    for i in range(len(s)):
        if s[i]=="(":
            stack.append(s[i])
        else:
            if stack:
                stack.pop()
            else:
                return False
    if stack:
        return False

    return True

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
class Solution {
    boolean solution(String s) {
        boolean answer = true;
        int cnt= 0;
        
        for(int i=0; i<s.length(); i++){
            if(s.charAt(i)=='('){
                cnt++;
            }
            else {
                if (cnt > 0){
                    cnt--;
                }
                else {
                    answer = false;
                    break;
                }
            }
        }
        if(cnt!=0) answer = false;
        return answer;
    }
}